### ISSUE TITLE
`modulo_arithmetic` should not trigger if result is only checked to be zero or non-zero

### ISSUE THREAD

**Issue by ronnodas** [rust-lang/rust-clippy#12006]
### Summary

A common usage of `%` is to check exact divisibility, and I believe `%` and `rem_euclid` output `0` for exactly the same set of inputs. So `a % b == 0` or `a % b != 0` should not trigger the lint.

### Lint Name

modulo_arithmetic

### Reproducer

I tried this code:

```rust
let x = 12_isize;
let y = -4_isize;
if x % y == 0{
    println!("{x} is divisible by {y}");
}
```

I saw this happen: The lint `modulo_arithmetic` triggers and suggests using `rem_euclid`.

I expected to see this happen: Nothing.

### Version

```text
rustc 1.74.1 (a28077b28 2023-12-04)
binary: rustc
commit-hash: a28077b28a02b92985b3a3faecf92813155f1ea1
commit-date: 2023-12-04
host: x86_64-unknown-linux-gnu
release: 1.74.1
LLVM version: 17.0.4
```
